lib/core/proxy.c

```diff
@@ -523,6 +530,14 @@ static h2o_httpclient_body_cb on_head(h2o_httpclient_t *client, const char *errs
         self->client = NULL;
         return NULL;
     }
+
+    if (forward_close_connection && conn_header != NULL) {
+        if (h2o_lcstris(conn_header->value.base, conn_header->value.len, H2O_STRLIT("close"))) {
+            if (self->src_req->version < 0x200)
+                self->src_req->http1_is_persistent = 0;
+        }
+    }
```

Comment:
Can we move the entire logic inside the loop?

I mean, we could do the check at the moment we find the `connection` header. Doing so would eliminate any overhead that we would have for http2client (because http2client in practice never emits a hop-by-hop header that we need to drop).

Reply:
SGTM. Makes perfect sense when considering H2.

Reply:
I've moved the logic to the parse loop in ba54b9a21